vaas/vaas/metrics/prometheus.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from datetime import timedelta
+from typing import Optional, TypedDict, Union
+from django.conf import settings
+from prometheus_client import CollectorRegistry, Gauge, Summary, push_to_gateway
+
+from vaas.metrics.models import Metrics
+
+logger = logging.getLogger(__name__)
+
+class MetricsBucket(TypedDict, total=False):
+    name: str
+    metric: Union[Summary, Gauge]
+
+
+class PrometheusClient:
+    def __init__(self, registry: CollectorRegistry) -> None:
+        self.host: str = f'{settings.PROMETHEUS_GATEWAY_HOST}:{settings.PROMETHEUS_GATEWAY_PORT}'
+        self.job: str = settings.PROMETHEUS_GATEWAY_JOB
+        self.registry: CollectorRegistry = registry
+
+    def push(self) -> None:
+        try:
+            push_to_gateway(gateway=self.host, job=self.job, registry=self.registry)
+        except Exception as e:
+            logger.exception(f'PrometheusClient: cannot push metric to vmagent')
+            pass
+
+
+class PrometheusMetrics(Metrics):
+    def __init__(self) -> None:
+        self.metrics_bucket: MetricsBucket = {}
+        self.registry: CollectorRegistry = CollectorRegistry()
+        super().__init__(
+            client=PrometheusClient(self.registry))
+
+    def _get_or_create(self, name: str, type: Union[Summary, Gauge]) -> Optional[Union[Summary, Gauge]]:
```

Comment:
Don't you think that this logic better fits to `PrometheusClient`? 
Currently, there is a shared internal object of type `CollectorRegistry` which is modified by `PrometheusMetrics` and read by `PrometheusClient`, it does not look well.

Reply:
Right. I currently move it to the client's responsibility.